contribute/design/1.x/cstor_cr_schema/apis.md

```diff
@@ -0,0 +1,550 @@
+
+	// OverProvisioning to enable over provisioning
+	// Optional -- defaults to false
+	OverProvisioning bool `json:"overProvisioning"`
+	// Compression to enable compression
+	// Optional -- defaults to off
+	// Possible values : lz, off
+	Compression string `json:"compression"`
+	// Resources are the compute resources required by the cstor-pool
+	// container.
+	Resources *corev1.ResourceRequirements `json:"resources"`
+	// AuxResources are the compute resources required by the cstor-pool pod
+	// side car containers.
+	AuxResources *corev1.ResourceRequirements `json:"auxResources"`
+	// Tolerations, if specified, the pool pod's tolerations.
+	Tolerations []corev1.Toleration `json:"tolerations"`
+
+	// PriorityClassName if specified applies to this pool pod
+	// If left empty, DefaultPriorityClassName is applied.
+	// (See CStorPoolClusterSpec.DefaultPriorityClassName)
+	// If both are empty, not priority class is applied.
+	PriorityClassName string `json:"priorityClassName"`
+}
+
+// RaidGroup contains the details of a raid group for the pool
+type RaidGroup struct {
+	// Type is the raid group type
+	// Supported values are : stripe, mirror, raidz and raidz2
+
+	// stripe -- stripe is a raid group which divides data into blocks and
+	// spreads the data blocks across multiple block devices.
+
+	// mirror -- mirror is a raid group which does redundancy
+	// across multiple block devices.
+
+	// raidz -- RAID-Z is a data/parity distribution scheme like RAID-5, but uses dynamic stripe width.
+	// radiz2 -- TODO
+	// Optional -- defaults to `defaultRaidGroupType` present in `PoolConfig`
+	Type string `json:"type"`
+
+	// LoadType can have following values :
+	// readCache -- The raid group is a read cache.
+	// writeCache -- The raid group is a write cache.
+	// spare -- The raid group is spare.
+	// "" -- Empty value means this will be used a regular vdev to form pool.
+	// What about default value ? Can this be changed after some point of time.
+	LoadType string `json:"load_type"`
+	// BlockDevices contains a list of block devices that
+	// constitute this raid group.
+
+	// DiskCount applies to raidz1 and raidz2 topology only.
+	// For raidz1 a valid DiskCount value is 2^n +1 where n>0
+	// For raidz2 a valid diskCount value is 2^n+2 where n>1
+	DiskCount int `json:"disk_count"`
+
+	BlockDevices []CStorPoolClusterBlockDevice `json:"blockDevices"`
+}
+
+// CStorPoolClusterBlockDevice contains the details of block devices that
+// constitutes a raid group.
+type CStorPoolClusterBlockDevice struct {
+	// BlockDeviceName is the name of the block device.
+	BlockDeviceName string `json:"blockDeviceName"`
+	// Capacity is the capacity of the block device.
+	// It is system generated
+	Capacity string `json:"capacity"`
+	// DevLink is the dev link for block devices
+	DevLink string `json:"devLink"`
+}
+
+// CStorPoolClusterStatus represents the latest available observations of a CSPC's current state.
+type CStorPoolClusterStatus struct {
```

Comment:
Status as:
```
type Status struct {
CurrentProvisionedInstances
RunningInstances
HealthyInstances
Conditions
}
```

Reply:
Will we have status like 
with 2 provisioned, one in init or degraded state, on in heathy state : 
```sh
NAME                         CURRENTPROVISIONEDINSTANCES   RUNNINGINSTANCES   HEALTHYINSTANCES   AGE
sparse-pool-1                       2/3                          1         	   1            142M
```
which will finally converge into all CSPI being Healthy as : 
```sh
NAME                         CURRENTPROVISIONEDINSTANCES   RUNNINGINSTANCES   HEALTHYINSTANCES   AGE
sparse-pool-1                       3/3                          3         	   3            155M
```

Reply:
looks good @shubham14bajpai

Reply:
@sonasingh46 can we make it as string as per above comment?

Reply:
otherwise, lets leave it like this only @sonasingh46 .. becz, string would be difficult to parse

Reply:
the above outputs are very confusing. what is the difference between running and healthy?

Reply:
Healthy gives the count of pools pods where pools are imported and are Healthy as well. RunningInstances gives the count of pool pods where pools are imported and are in states like Healthy/Degraded etc.

Reply:
`3/3`  kind of arrangement cannot be done as we do not have control of kubectl.
One workaround would be to convert the field into a string but that will end up into too much parsing.
So `/` thing may not be included. I am opting out of this inclusion.
But this can be an item to track -- e.g. if some plugins in kubectl etc can help to extend.